ngraph/core/src/op/dft.cpp

```diff
@@ -0,0 +1,259 @@
+    {
+        const auto input_rank = input_shape.rank().get_length();
+        const auto& const_axes = get_constant_from_source(input_value(1));
+        auto axes = const_axes->cast_vector<int64_t>();
+
+        for (int64_t& axis : axes)
+        {
+            if (axis < 0)
+            {
+                axis += input_rank - 1;
+            }
+        }
+
+        AxisVector axes_vector;
+        AxisSet axes_set;
+        for (const int64_t axis : axes)
+        {
+            axes_vector.push_back(static_cast<size_t>(axis));
+            axes_set.insert(static_cast<size_t>(axis));
+        }
+
+        NODE_VALIDATION_CHECK(
+            this, axes.size() == axes_set.size(), "DFT axes must be unique. Got: ", axes_vector);
+
+        NODE_VALIDATION_CHECK(this,
+                              std::find(axes.begin(), axes.end(), input_rank - 1) == axes.end(),
+                              "DFT axes cannot contain the last axis. Got axes: ",
+                              axes_vector);
+    }
+
+    if (num_of_inputs == 3)
+    {
+        element::Type signal_size_et = get_input_element_type(2);
+        NODE_VALIDATION_CHECK(this,
+                              signal_size_et == element::i64 || signal_size_et == element::i32,
+                              "DFT signal_size element type must be i32 or i64");
+
+        PartialShape signal_size_shape = PartialShape(get_input_partial_shape(2));
+        if (signal_size_shape.rank().is_static())
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  signal_size_shape.rank().get_length() == 1,
+                                  "DFT Signal size input must be 1D tensor. Got signal size "
+                                  "input rank: ",
+                                  signal_size_shape.rank().get_length());
+        }
+
+        if (axes_shape.is_static() && signal_size_shape.is_static())
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  axes_shape.to_shape()[0] == signal_size_shape.to_shape()[0],
+                                  "Sizes of inputs 'axes' and 'sinal_size' must be equal. Got "
+                                  "size of 'axes': ",
+                                  axes_shape.to_shape()[0],
+                                  "size of 'signal_size': ",
+                                  signal_size_shape.to_shape()[0]);
+        }
+    }
+}
+
+void op::v7::DFT::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v7_DFT_validate_and_infer_types);
+    validate();
+
+    PartialShape input_shape = PartialShape(get_input_partial_shape(0));
+    PartialShape axes_shape = PartialShape(get_input_partial_shape(1));
+    PartialShape output_shape = input_shape;
+    if (input_shape.rank().is_dynamic() || axes_shape.rank().is_dynamic())
+    {
+        set_output_type(0, get_input_element_type(0), output_shape);
+        return;
+    }
```

Comment:
Consider the following example of the DFT with the following input shapes: `DFT([10, 20, 2], Dynamic, [2])`. According to the current implementation the output shape will be `[10, 20, 2]`, but this is wrong, because we may pad or cut-off some of the values based on the values specified in the third input! The output shape in this case should be `[Dynamic, Dynamic, 2]`.

Reply:
Please, add such a unit test.

Reply:
If the input from port 1 has dynamic shape, then this input is not `Constant`, hence, we will have the output shape `[Dynamic, Dynamic, 2]`.

Reply:
Provided explanation in the internal chat

Reply:
Done.